experimental/material_3_demo/lib/main.dart

```diff
@@ -176,40 +251,393 @@ class _Material3DemoState extends State<Material3Demo> {
       title: 'Material 3',
       themeMode: useLightMode ? ThemeMode.light : ThemeMode.dark,
       theme: themeData,
-      home: LayoutBuilder(builder: (context, constraints) {
-        if (constraints.maxWidth < narrowScreenWidthThreshold) {
-          return Scaffold(
-            appBar: createAppBar(),
-            body: Row(children: <Widget>[
-              createScreenFor(ScreenSelected.values[screenIndex], false),
-            ]),
-            bottomNavigationBar: NavigationBars(
-              onSelectItem: handleScreenChanged,
-              selectedIndex: screenIndex,
-              isExampleBar: false,
-            ),
-          );
-        } else {
-          return Scaffold(
-            appBar: createAppBar(),
-            body: SafeArea(
-              bottom: false,
-              top: false,
-              child: Row(
-                children: <Widget>[
-                  Padding(
-                      padding: const EdgeInsets.symmetric(horizontal: 5),
-                      child: NavigationRailSection(
-                          onSelectItem: handleScreenChanged,
-                          selectedIndex: screenIndex)),
-                  const VerticalDivider(thickness: 1, width: 1),
-                  createScreenFor(ScreenSelected.values[screenIndex], true),
-                ],
+      home: AnimatedBuilder(
+          animation: controller,
+          builder: (context, child) {
+            return NavigationTransition(
+              scaffoldKey: scaffoldKey,
+              animationController: controller,
+              railAnimation: railAnimation,
+              appBar: createAppBar(),
+              body: createScreenFor(
+                  ScreenSelected.values[screenIndex], controller.value == 1),
+              navigationRail: NavigationRail(
+                extended: showLargeSizeLayout,
+                destinations: navRailDestinations,
+                selectedIndex: screenIndex,
+                onDestinationSelected: (index) {
+                  setState(() {
+                    screenIndex = index;
+                    handleScreenChanged(screenIndex);
+                  });
+                },
+                trailing: Expanded(
+                    child: Padding(
+                  padding: const EdgeInsets.only(bottom: 20),
+                  child: showLargeSizeLayout
```

Comment:
For readability, can you wrap these subtrees in methods that return a Widget?

As in

```dart
child: showLargeSizeLayout ? _largeLayout() : _smallLayout()
```

Reply:
Fixed:)